controller/index.js

```diff
@@ -1,1019 +1,1054 @@
+    if (req.params.layer) {
+      req.query.layer = req.params.layer
     }
 
-    agol.find(req.params.id, function(err, data){
+    agol.find(req.params.id, function (err, data) {
       if (err) {
-        res.status(err.code || 404).send( err );
-      } else {
-        // Get the item 
-        agol.dropItem( data.host, req.params.item, req.query, function(error, itemJson){
-          if (error) {
-            res.status(error.code || 400).send( error );
-          } else {
-            res.json( itemJson );
-          }
-        });
+        return res.status(err.code || 404).send(err)
       }
-    });
-  };
-
-  // find the items data 
-  controller.findItemData = function(req, res){
-    var self = this;
-    // closure that actually goes out gets the data
-    var _get = function(id, item, key, options, callback){
-       agol.find( id, function( err, data ){
-        if (err) {
-          callback(err, null);
-        } else {
-          // Get the item
-          if ( !parseInt(options.layer) ){
-            options.layer = 0;
-          }
 
-          agol.getItemData( data.host, id, item, key, options, function(error, itemJson){
-            if (error) {
-              callback( error, null);
-            // if we have status return right away
-            } else if (itemJson.koop_status == 'processing' && typeof req.params.silent === 'undefined') {
-              // return w/202
-              agol.getCount(['agol', item, options.layer].join(':'), {}, function(err, count) {
-                var code = 202;
-                var response = {
-                  status: 'processing',
-                  processing_time: ( Date.now() - itemJson.retrieved_at)/1000 || 0, 
-                  count: count
-                };
-                if ( itemJson.generating ){
-                  response.generating = itemJson.generating;
-                  // we received an error from the server
-                  if (itemJson.generating.error){
-                    code = 502;
-                  }
-                }
-                res.status( code ).json( response );
-              }); 
-            } else {
-              callback( null, itemJson );
-            }
-          });
+      // Get the item
+      agol.dropItem(data.host, req.params.item, req.query, function (error, itemJson) {
+        if (error) {
+          return res.status(error.code || 400).send(error)
+        }
+        res.json(itemJson)
+      })
+    })
+  }
+
+  /**
+   * Gets the actual json data from the model
+   *
+   * @param {object} params - required params used to fetch data
+   * @param {object} options - optional query string based params
+   * @param {function} callback
+   */
+  controller._getItemData = function (req, res, callback) {
+    var params = req.params
+    var options = req.query
+    var id = params.id
+    var item = params.item
+    var key = params.key
+
+    agol.find(id, function (err, data) {
```

Comment:
seems like this logic should only happen in one place since most incoming requests will need to do it